pkg/task/task.go

```diff
@@ -0,0 +1,61 @@
+package task
+
+import "time"
+
+const (
+	TaskStateRequested TaskState = iota // when client requests a task
+	TaskStateScheduled                  // daemon accepts the task into the queue
+	TaskStateBuilding                   // daemon is building the task
+	TaskStateRunning                    // daemon is running the task
+	TaskStateComplete                   // daemon is not running the task. ready to collect.
+)
+
+type TaskState int
+
+func (t TaskState) String() string {
+	return [...]string{
+		"requested",
+		"scheduled",
+		"building",
+		"running",
```

Comment:
For the purposes of a task queue utility, all it should be knowing is that it's running. It should stay oblivious to the actual meaning of the job.